linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -1150,14 +1150,20 @@ void ApplyPreserveMethods (MethodDefinition method)
 			MarkMethodCollection (list);
 		}
 
-		protected void MarkFields (TypeDefinition type, bool includeStatic)
+		protected void MarkFields (TypeDefinition type, bool includeStatic, bool markBackingFieldsOnlyIfPropertyMarked = false)
 		{
 			if (!type.HasFields)
 				return;
 
 			foreach (FieldDefinition field in type.Fields) {
 				if (!includeStatic && field.IsStatic)
 					continue;
+				if (markBackingFieldsOnlyIfPropertyMarked && field.Name.EndsWith ("k__BackingField")) {
+					var propertyName = field.Name.Substring (1, field.Name.IndexOf ('>') - 1);
+					var propertyDefinition = type.Properties.First (p => p.Name == propertyName);
```

Comment:
Yes, compiler could reuse this name for something else or some other tool could generated only field with the name.

Also `field.Name.IndexOf ('>')` check fails with code like

```
interface I<T>
{
	int Foo { get; set; }
}

class X : I<int>
{
	int I<int>.Foo { get; set; }
}

```